docs/source/tutorials/building_blocks.rst

```diff
@@ -0,0 +1,342 @@
+       def construct(self):
+           circle = Circle().shift(LEFT)
+           square = Square().shift(UP)
+           triangle = Triangle().shift(RIGHT)
+
+           circle.set_stroke(color=GREEN, width=20)
+           square.set_fill(YELLOW, opacity=1.0)
+           triangle.set_fill(PINK, opacity=0.5)
+
+           self.add(triangle, square, circle)
+           self.wait(1)
+
+The only difference here (besides the scene name) is the order in which the
+mobjects are added to the scene.  In ``MobjectStyling``, we added them as
+``add(circle, square, triangle)``, whereas in ``MobjectZOrder`` we add them as
+``add(triangle, square, circle)``.  The result is the following:
+
+.. image:: ../_static/building_blocks/zorder.png
+    :align: center
+    :alt: figure of a circle, a square, and a triangle
+
+As you can see, the order of the arguments of :meth:`~.Scene.add` determines
+the order that the mobjects are displayed on screen, with the left-most
+arguments being put in the back.
+
+
+**********
+Animations
+**********
+
+At the heart of manim is animation.  Generally, you can add an animation to
+your scene by calling the :meth:`~.Scene.play` method.
+
+.. code-block:: python
+
+   class SomeAnimations(Scene):
+       def construct(self):
+           square = Square()
+           self.add(square)
+
+	   # some animations display mobjects, ...
+	   self.play(FadeIn(square))
+
+	   # ... some move or rotate mobjects around...
+	   self.play(Rotate(square, PI/4))
+
+	   # some animations remove mobjects from the screen
+	   self.play(FadeOut(square))
+
+           self.wait(1)
+
+.. image:: ../_static/building_blocks/some_animations.gif
+    :align: center
+    :alt: a square fading it, rotating, and fading out
+
+Put simply, animations are procedures that interpolate between two mobjects.
+For example, :code:`FadeIn(square)` starts with a fully transparent version of
+:code:`square` and ends with a fully opaque version, interpolating between them
+by gradually increasing the opacity.  :class:`.FadeOut` works in the opposite
+way: it interpolates from fully opaque to fully transparent.  As another
+example, :class:`.Rotate` starts with the mobject passed to it as argument, and
+ends with the same object but rotated by a certain amount, this time
+interpolating the mobject's angle instead of its opacity.
+
+
+Animating methods
+=================
+
+Any property of a mobject that can be changed can be animated.  In fact, any
+method that changes a mobject's property can be used as an animation, through
+the use of :class:`.ApplyMethod`.
+
+.. code-block:: python
+
+   class ApplyMethodExample(Scene):
+       def construct(self):
+           square = Square().set_fill(RED, opacity=1.0)
+           self.add(square)
+
+	   # animate the change of color
+	   self.play(ApplyMethod(square.set_fill, WHITE))
+	   self.wait(1)
+
+	   # animate the change of position
+	   self.play(ApplyMethod(square.shift, UP))
+           self.wait(1)
```

Comment:
The indentation is off here (and I am not confident enough in my counting skills to make a suggestion, sorry 🙃)

Reply:
Good eye! Just fixed that in the latest commit.